### DISCUSSION TITLE
Akvorado Inlet Internal Queues

### DISCUSSION THREAD

**Discussion by adjiekrazz** [akvorado/akvorado#1507] [Q&A] [answered]
Hello @vincentbernat and others,

I have several questions about [Inlet Internal Queues](https://github.com/akvorado/akvorado/blob/main/inlet/flow/input/udp/config.go#L17) below: 
```
// Configuration describes UDP input configuration.
type Configuration struct {
	...
	// QueueSize defines the size of the channel used to
	// communicate incoming flows. 0 can be used to disable
	// buffering.
	QueueSize uint
	...
}
```

1. Can we change the QueueSize value via orchestrator yaml file?
2. Can we get and monitor the QueueSize metric through Inlet's prometheus exporter?
3. What we can do if the QueueSize is reached its limit? i have tried to add more workers for this but it seems that the queue is separated between each workers.

The only metric that i can collect is `akvorado_inlet_flow_input_udp_out_dropped_packets_total` that keep increased after QueueSize is full.  Here is the error logs:

> {"level":"warn","worker":"7","listen":":2055","time":"2024-11-18T07:56:33Z","caller":"akvorado/inlet/flow/input/udp/root.go:207","module":"akvorado/inlet/flow/input/udp","message":"dropping flow due to queue full (size 100000)"}

Thanks in advance!

**Comment by adjiekrazz** [comment]
After reread the docs, i already found the answer for questions number 1 and 3 in this [docs](https://demo.akvorado.net/docs/configuration#flow).  But i have another question about how we know the best number of workers configuration for inlet based on its hardware specs? for example in the ClickHouse consumer, it depends on num of CPU cores.  1 worker uses 1 core cpu.  How about inlet?

**Reply by vincentbernat** [reply, marked as answer]
It is not possible to monitor the queue size, you can only check when queue is full. Workers are implemented with goroutines, so there is no 1:1 mapping. When everything runs smoothly, workers should be CPU-bound but when for example you have some issues with Kafka, they may wait for Kafka (which has its own internal queue).
